### PR TITLE
[INT-442] Poetry

### PR DESCRIPTION
**Pull request by bbonenfant**
Switch from `setup.py` to `poetry`/`pyproject.toml`

This changes the following things:
* No more need for a `setup.py` file, so it gets deleted. Building the wheel is done through `poetry build` and the the dependencies are tracked with `pyproject.toml` and `poetry.lock`
* There's also no need for the `MANIFEST.in` file as `poetry` does all of that in the `pyproject.toml` file. There should probably be a review of this to ensure that the proper things get included in the package that were previously specified in the manifest.
* `poetry` installs the virtual environment at `.venv` instead of `venv`. No way I know around this unfortunately.
* I switched how the `__version__` value gets specified. The modern way to do this is to look at the metadata of the installed package -- this functionality is built into the language in python >=3.8, but there is backport (`importlib-metadata`) for older versions.
    * On this point, `poetry` _must_ have the version specified in the `pyproject.toml` so there should probably be a discussion on how this meshes with the`version.json` file.
* Had to make some tweaks to get CI working (but surprisingly fewer tweaks than I was expecting)
* I changed the `author` of the package on whim to `Pachyderm <python-pachyderm@pachyderm.io>`, but I can revert to JD if that was a mistake.

Follow ons:
* There should be a review and sanitization of the dependency pinning. In general testing dependencies should be pinned and the `^` dependency selector should be used over `>/=>` for package dependencies.

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -2,11 +2,18 @@
...
 
+
+Once both `poetry` is installed, you can use `poetry` to create your
+  virtual environment for this project. The following command
+  (run at the root of this project) will create a virtual environment
+  within the `.venv` directory:
 ```bash
-python3 -m venv venv
-source venv/bin/activate
+poetry install
```
Comment (albscui) [inline contributing.md:15]: Does this install the dev dependencies as well?

**Review by albscui — APPROVED** [review]
LGTM!

Although I don't know poetry that well, this looks really useful!

File: contributing.md (hunk 1 above)
Comment (bbonenfant) [reply contributing.md:15]: Yep! When run like this on the repo `poetry` will install all dependencies, including dev dependencies.

**Comment by smalyala** [comment]
Just to confirm, if we want to bump the python-pachyderm version, we need to update it in `pyproject.toml` and `version.json`. As of right now, it seems like the python-pachyderm version in `version.json` is only being used in setup.py (and in `test_version.py`). Correct me if I'm wrong, but we can effectively remove the python-pachyderm line from `version.json` and instead use the version number from `pyproject.toml` for version tests right?

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,55 @@
+[tool.poetry]
+name = "python-pachyderm"
+version = "7.2.0"
+description = "Python Pachyderm Client"
+authors = ["Pachyderm <python-pachyderm@pachyderm.io>"]
```
Comment (smalyala) [inline pyproject.toml:5]: Let's use the email `integrations@pachyderm.io` instead. @albscui thoughts?

File: src/python_pachyderm/__init__.py [hunk 3]
```diff
@@ -15,9 +15,16 @@
 from grpc import RpcError
 from .experimental import ExperimentalClient
 
+# Python version compatibility.
+try:
+    # >= 3.8
+    import importlib.metadata as metadata
```
Comment (smalyala) [inline src/python_pachyderm/__init__.py:21]: Does this read directly from `pyproject.toml`?

File: pyproject.toml (hunk 2 above)
Comment (bbonenfant) [reply pyproject.toml:5]: I grabbed this from [this issue](https://github.com/pachyderm/python-pachyderm/issues/280) on a whim. Happy to change it to the correct one

File: src/python_pachyderm/__init__.py (hunk 3 above)
Comment (bbonenfant) [reply src/python_pachyderm/__init__.py:21]: Nope, packages keep their official version in an associated `METADATA` file that these modules attempt to read (this is how pip can report the version of installed packages without a lock file).

File: poetry.toml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[virtualenvs]
+in-project = true
```
Comment (smalyala) [inline poetry.toml:2]: What does this file do?

Comment (bbonenfant) [reply poetry.toml:2]: `poetry` by default tries to create all you virtual environments in your home directory, this just tells poetry to create the virtual environment in the project

Comment (smalyala) [reply poetry.toml:2]: Neat!

**Review by smalyala — APPROVED** [review]
LGTM, just one question up above

File: pyproject.toml (hunk 2 above)
Comment (smalyala) [reply pyproject.toml:5]: Great! This integrations email is the one we're using for managing all integrations-related projects moving forward.

**Pull request merged**